app/views/tcsp/summary.scala.html

```diff
@@ -38,14 +39,24 @@
     ) {
         @model.tcspTypes.map { v =>
         <ul class="list list--semicolon">
-            @v.serviceProviders.map { x =>
-                <li>@x match {
-                    case NomineeShareholdersProvider => {@Messages("tcsp.service.provider.lbl.01")}
-                    case TrusteeProvider => {@Messages("tcsp.service.provider.lbl.02")}
-                    case RegisteredOfficeEtc => {@Messages("tcsp.service.provider.lbl.03")}
-                    case CompanyDirectorEtc => {@Messages("tcsp.service.provider.lbl.04")}
-                    case CompanyFormationAgent(_, _) => {@Messages("tcsp.service.provider.lbl.05")}
-                }</li>
+            @if(v.serviceProviders.contains(CompanyDirectorEtc)) {
```

Comment:
There is probably a cleverer way to do this, but definitely not as clear. I like it 👍

Reply:
use list.sortedBy please!